### PR TITLE
Add support for Zipper.find_all/3

### PR DESCRIPTION
**Pull request by bcardarella**

### CODE REVIEW

File: test/zipper_test.exs [hunk 1]
```diff
@@ -571,6 +571,72 @@ defmodule SourcerorTest.ZipperTest do
...
+      zipper =
+        [1, [2, [3, 4], 5]]
+        |> Z.zip()
+        |> Z.next()
+        |> Z.next()
+        |> Z.next()
+        |> Z.next()
+        |> Z.next()
+        |> Z.next()
+        |> Z.next()
```
Comment (bcardarella) [inline test/zipper_test.exs:615]: if there is a better way to get to the last node let me know. I tried `Zipper.rightmost` but I think that only goes to the right most of the current node

**Comment by bcardarella** [comment]
FYI unrelated tests failing with Elixir 1.17

```
  1) test get_range/1 bitstrings (SourcerorTest.RangeTest)
     test/range_test.exs:1041
     Assertion with == failed
     code:  assert decorate(code, to_range(code)) == "«<<1, 2,\n\n foo>>»\n" |> String.trim_trailing()
     left:  "«<<1, 2,\n\n foo>>\n»"
     right: "«<<1, 2,\n\n foo>>»"
     stacktrace:
       test/range_test.exs:1051: (test)

......

  2) test get_range/1 anonymous functions (SourcerorTest.RangeTest)
     test/range_test.exs:664
     Assertion with == failed
     code:  assert decorate(code, to_range(code)) == "«fn ->\n  :ok\nend»\n" |> String.trim()
     left:  "«fn ->\n  :ok\nend\n»"
     right: "«fn ->\n  :ok\nend»"
     stacktrace:
       test/range_test.exs:674: (test)

................................................................................................................................................................

  3) test get_end_position/2 returns the correct positions (SourcerorTest)
     test/sourceror_test.exs:546
     Assertion with == failed
     code:  assert Sourceror.get_end_position(quoted) == [line: 3, column: 1]
     left:  [line: 3, column: 3]
     right: [line: 3, column: 1]
     stacktrace:
       test/sourceror_test.exs:565: (test)

.

  4) doctest Sourceror.get_end_position/2 (20) (SourcerorTest)
     test/sourceror_test.exs:7
     Doctest failed
     doctest:
       iex> quoted = ~S"""
       ...> foo do
       ...>   :ok
       ...> end
       ...> """ |>  Sourceror.parse_string!()
       iex> Sourceror.get_end_position(quoted)
       [line: 3, column: 1]
     code:  Sourceror.get_end_position(quoted) === [line: 3, column: 1]
     left:  [line: 3, column: 3]
     right: [line: 3, column: 1]
     stacktrace:
       lib/sourceror.ex:562: Sourceror (module)
```

**Comment by bcardarella** [comment]
force pushed format fix for CI

**Review by doorgan — APPROVED** [review]
Looks great, thank you!

About the 1.17 test issues, it seems some bugs in the elixir parser were fixed in core elixir. We can either backport those fixes in a new PR, or adopt [lib_elixir](https://github.com/zachallaun/lib_elixir)(I have a POC [here](https://github.com/doorgan/sourceror/pull/162) but there's some polishing needed to make it work appropriately)

**Pull request merged**